content/hardware/04.pro/boards/portenta-x8/tutorials/16.edge-ai-docker-container/content.md

```diff
@@ -0,0 +1,1145 @@
+
+By prioritizing the most important features, the model can be updated to be more efficient and accurate in detecting deviations, reducing the risk of false positives and ensuring robust real time anomaly classification.
+
+With feature extraction completed, the system is now ready for model training. The extracted features will be used to train a classification model that distinguishes between different flow states.
+
+***Please refer to the [Edge Impulse Sensor Fusion Documentation](https://docs.edgeimpulse.com/docs/tutorials/end-to-end-tutorials/time-series/sensor-fusion) for further insights into DSP based sensor fusion and anomaly detection.***
+
+### Model Training
+
+Using Edge Impulse, the processed data is used to train a machine learning model capable of classifying flow conditions. The objective of the model is to:
+
+- Recognize normal flow patterns based on historical data.
+- Detect anomalous flow conditions, such as unexpected fluctuations, obstructions or irregular flow rates.
+
+Different machine learning techniques, such as decision trees, neural networks or anomaly detection models, may be considered to determine the most effective approach. This project uses a neural network classifier to map extracted spectral features to flow states. The following image shows the *Neural Network settings and training process*:
+
+![Neural Network settings and training process](assets/edge-impulse-classifier-train-result.png)
+
+**Neural Network Architecture** is configured with:
+
+- **Input layer:** 21 features extracted from spectral analysis.
+- **Hidden layers:** 30 neurons (Dense Layer) / 10 neurons (Dense Layer) / 15 neurons (Dense Layer)
+- **Output layer:** 4 classes (Normal, Low, High, Anomalous)
+
+The network uses *ReLU activation functions* in hidden layers and *Softmax activation* in the output layer to predict the most probable class.
+
+The Neural Network Training Configuration is as follows:
+
+| **Parameter**           | **Value**                    |
+|-------------------------|------------------------------|
+| **Training cycles**     | 150                          |
+| **Learning rate**       | 0.0005                       |
+| **Processor**           | CPU based training           |
+| **Validation set size** | 30% of the dataset           |
+| **Batch size**          | 32 (not enabled in this run) |
+
+
+The model's *quantized (int8) version* is used for deployment to optimize inference speed and reduce memory usage.
+
+#### Performance Metrics
+
+Once training is complete, the model's accuracy is evaluated using a confusion matrix and performance scores. The key results include:
+
+| **Metric**                     | **Value** |
+|--------------------------------|-----------|
+| **Overall Model Accuracy**     | 73.3%     |
+| **Weighted Average Precision** | 0.87      |
+| **Weighted Average Recall**    | 0.73      |
+| **Weighted Average F1 Score**  | 0.77      |
+| **Area Under ROC Curve (AUC)** | 0.86      |
```

Comment:
coooooooooooooool

Reply:
Updated with addition of following note for technical reference of each metric:

For a detailed explanation of each metric, please refer to the [evaluation metrics documentation from Edge Impulse](https://docs.edgeimpulse.com/docs/concepts/machine-learning/evaluation-metrics).